contracts/contracts/GrantRegistry.sol

```diff
@@ -0,0 +1,135 @@
+pragma solidity ^0.8.5;
+
+contract GrantRegistry {
```

Comment:
The goal of the GrantRegistry is to allow anyone to create a Grant. 

Our options as I see them are as follows:
1) Have a dApp deploy a new contract for each grant that is created
2) Have a factory pattern contract that deploys a new grant contract for each new grant created
3) Have a contract that follows a pattern similar to the multi token standard (ERC1155) where grants are created and tracked within a single GrantRegistry contract

Option #3 seems the most cost effective to me but it would be great to see a more data informed comparison between our options. In this first round of prototype, it may be worth simply implementing #3 unless we have a strong reason not to and reassess if needed.

After our discussion this morning, I think we can move forward assuming that there will be one GrantRegistry contract that we will support. Anyone could of course deploy their own GrantRegistry contract but that would result in a completely separate ecosystem.

This GrantRegistry would hold ALL grants and not be filtered or curated. A grant would and could only ever exist in one GrantRegistry.

A GrantRound would have a subset of grants from a GrantRegistry. The list could live on or off chain. 

If on-chain, it could be a list contract which contains an array of grants. The GrantRound would link to the List contract to define which grants are in the round.

If off-chain, the GrantRound could link to off chain metadata which might be a JSON document containing an array of grants.

The lists could be created manually but I imagine would be eventually be the result of first curation (removing the 'bad' grants through some process, or including the 'good' grants using properties like sybil score and the entities participation policy), then applying a filter (selecting grants with the desired properties such as theme, capital raised, etc...)